File: Contribute/Plugin Samples/Plugin Samples/VB.NET/CL HighlightingPluginSample/MainForm.vb
'CL Syntax Highlighter Plugin

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports WeifenLuo.WinFormsUI.Docking
Imports SS.Ynote.Engine.Framework.Plugins.Interface
Imports System.Xml.Linq
Imports FastColoredTextBoxNS
Imports SS.Ynote.Classic
Imports System.Windows.Forms

Public Class MainForm
	Inherits DockContent
	Implements IFormPlugin
	#Region "Constructor"

	Public Sub New()
		InitializeComponent()
		Me.ShowHint = DockState.DockLeft
	End Sub
	#End Region

	Private button1 As Button
	Private button2 As Button

	#Region "IPluginMembers"

	'IFormPlugin Inherits the IPlugin Interface

	Public ReadOnly Property Title() As String Implements IPlugin.Title
		Get
			Return "HighlighterPlugin"
		End Get
	End Property
	Public ReadOnly Property Description() As String Implements IPlugin.Description
		Get
			Return "Sample Highlighter Plugin"
		End Get
	End Property
	Public ReadOnly Property Group() As String Implements IPlugin.Group
		Get
			Return "Plugins"
		End Get
	End Property
	Public ReadOnly Property SubGroup() As String Implements IPlugin.SubGroup
		Get
			Return "Samples"
		End Get
	End Property

	Private m_configuration As New XElement("HighlighterPluginConfig")
	Public Property Configuration() As XElement Implements IPlugin.Configuration
		Get
			Return m_configuration
		End Get
		Set
			m_configuration = value
		End Set
	End Property

	Public ReadOnly Property Icon() As String Implements IPlugin.Icon
		Get
			Return String.Empty
		End Get
	End Property

	#End Region

	#Region "IFormPlugin Members"

	Public ReadOnly Property Content() As DockContent Implements IFormPlugin.Content
		Get
			Return Me
		End Get
	End Property
	Public ReadOnly Property ShowAs() As ShowAs Implements IFormPlugin.ShowAs
		Get
			Return ShowAs.Normal
		End Get
	End Property
	#End Region

	Private Sub ActiveFastColoredTextBox_TextChangedDelayed(sender As Object, e As TextChangedEventArgs)
		Dim [Custom] As New List(Of ExplorerItem)()
		HighlightingPluginSample.CLSyntaxHighlighter.Highlight(e.ChangedRange, [Custom])
	End Sub

	Private Sub InitializeComponent()
		Me.button1 = New System.Windows.Forms.Button()
		Me.button2 = New System.Windows.Forms.Button()
		Me.SuspendLayout()
		' 
		' button1
		' 
		Me.button1.Dock = System.Windows.Forms.DockStyle.Fill
		Me.button1.Font = New System.Drawing.Font("Microsoft Sans Serif", 50F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CByte(0))
		Me.button1.Location = New System.Drawing.Point(0, 0)
		Me.button1.Name = "button1"
		Me.button1.Size = New System.Drawing.Size(358, 257)
		Me.button1.TabIndex = 0
		Me.button1.Text = "Highlight"
		Me.button1.UseVisualStyleBackColor = True
		AddHandler Me.button1.Click, New System.EventHandler(AddressOf Me.button1_Click)
		' 
		' button2
		' 
		Me.button2.Dock = System.Windows.Forms.DockStyle.Bottom
		Me.button2.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25F)
		Me.button2.Location = New System.Drawing.Point(0, 257)
		Me.button2.Name = "button2"
		Me.button2.Size = New System.Drawing.Size(358, 249)
		Me.button2.TabIndex = 1
		Me.button2.Text = "Insert Sample Text"
		Me.button2.UseVisualStyleBackColor = True
		AddHandler Me.button2.Click, New System.EventHandler(AddressOf Me.button2_Click)
		' 
		' MainForm
		' 
		Me.ClientSize = New System.Drawing.Size(358, 506)
		Me.Controls.Add(Me.button1)
		Me.Controls.Add(Me.button2)
		Me.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CByte(0))
		Me.Name = "MainForm"
		Me.ShowInTaskbar = False
		Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
		Me.Text = "OpenCL Highlighter Plugin"
		AddHandler Me.Load, New System.EventHandler(AddressOf Me.MainForm_Load)
		Me.ResumeLayout(False)

	End Sub

	Private Sub MainForm_Load(sender As Object, e As EventArgs)

	End Sub

	Private Sub button1_Click(sender As Object, e As EventArgs)
		Try
			AddHandler Main.ActiveFastColoredTextBox.TextChangedDelayed, New EventHandler(Of TextChangedEventArgs)(AddressOf ActiveFastColoredTextBox_TextChangedDelayed)
			MessageBox.Show("Highlighter Initialized. Just Type in Text To Highlight")
		Catch ex As Exception
			Console.WriteLine(ex.Message)
		End Try
	End Sub

	Private Sub button2_Click(sender As Object, e As EventArgs)
		Try
			Main.ActiveFastColoredTextBox.InsertText("kernel void Mono(" & vbLf & "_global uchar4 *input," & vbLf & "const float2 file" & vbLf & ")" & vbLf & vbLf & "int Main()" & vbLf & "{" & vbLf & "int coord=(int2)(get_global_id(0));" & vbLf & "}")
		Catch
		End Try
	End Sub
End Class

File: Contribute/Plugin Samples/C#/Highlighting/CL HighlightingPluginSample/MainForm.cs
//CL Syntax Highlighter Plugin

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using WeifenLuo.WinFormsUI.Docking;
using SS.Ynote.Engine.Framework.Plugins.Interface;
using System.Xml.Linq;
using FastColoredTextBoxNS;
using SS.Ynote.Classic;
using System.Windows.Forms;

namespace HighlightingPluginSample
{
    public class MainForm :  DockContent, IFormPlugin
    {
        #region Constructor

        public MainForm()
        {
            InitializeComponent();
            this.ShowHint = DockState.DockLeft;
        }
        #endregion

        private Button button1;
        private Button button2;

        #region IPluginMembers

        //IFormPlugin Inherits the IPlugin Interface

        public string Title
        {
            get { return "HighlighterPlugin"; }
        }
        public string Description
        {
            get { return "Sample Highlighter Plugin"; }
        }
        public string Group
        {
            get { return "Plugins"; }
        }
        public string SubGroup
        {
            get { return "Samples"; }
        }

        private XElement configuration = new XElement("HighlighterPluginConfig");
        public XElement Configuration
        {
            get { return configuration; }
            set { configuration = value; }
        }

        public string Icon
        {
            get { return string.Empty; }
        }

        #endregion

        #region IFormPlugin Members

        public DockContent Content
        {
            get { return this; }
        }
        public ShowAs ShowAs
        {
            get { return ShowAs.Normal; }
        }
        #endregion

        private void codebox_TextChangedDelayed(object sender, TextChangedEventArgs e)
        {
            List<ExplorerItem> Custom = new List<ExplorerItem>();
            HighlightingPluginSample.CLSyntaxHighlighter.Highlight(e.ChangedRange, Custom);
        }

        private void InitializeComponent()
        {
            this.button1 = new System.Windows.Forms.Button();
            this.button2 = new System.Windows.Forms.Button();
            this.SuspendLayout();
            // 
            // button1
            // 
            this.button1.Dock = System.Windows.Forms.DockStyle.Fill;
            this.button1.Font = new System.Drawing.Font("Microsoft Sans Serif", 50F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.button1.Location = new System.Drawing.Point(0, 0);
            this.button1.Name = "button1";
            this.button1.Size = new System.Drawing.Size(358, 257);
            this.button1.TabIndex = 0;
            this.button1.Text = "Highlight";
            this.button1.UseVisualStyleBackColor = true;
            this.button1.Click += new System.EventHandler(this.button1_Click);
            // 
            // button2
            // 
            this.button2.Dock = System.Windows.Forms.DockStyle.Bottom;
            this.button2.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F);
            this.button2.Location = new System.Drawing.Point(0, 257);
            this.button2.Name = "button2";
            this.button2.Size = new System.Drawing.Size(358, 249);
            this.button2.TabIndex = 1;
            this.button2.Text = "Insert Sample Text";
            this.button2.UseVisualStyleBackColor = true;
            this.button2.Click += new System.EventHandler(this.button2_Click);
            // 
            // MainForm
            // 
            this.ClientSize = new System.Drawing.Size(358, 506);
            this.Controls.Add(this.button1);
            this.Controls.Add(this.button2);
            this.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.Name = "MainForm";
            this.ShowInTaskbar = false;
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
            this.Text = "OpenCL Highlighter Plugin";
            this.Load += new System.EventHandler(this.MainForm_Load);
            this.ResumeLayout(false);

        }

        private void MainForm_Load(object sender, EventArgs e)
        {
           
        }

        private void button1_Click(object sender, EventArgs e)
        {
            try
            {
                Main.ActiveEditor.codebox.TextChangedDelayed += new EventHandler<TextChangedEventArgs>(codebox_TextChangedDelayed);
                MessageBox.Show("Highlighter Initialized. Just Type in Text To Highlight");
            }
            catch (Exception ex) { Console.WriteLine(ex.Message); }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            try
            {
                Main.ActiveEditor.codebox.InsertText("kernel void Mono(\n_global uchar4 *input,\nconst float2 file\n)\n\nint Main()\n{\nint coord=(int2)(get_global_id(0));\n}");
            }
            catch { }
        }
    }
}
